src/storage/storage-driver-local-storage.ts

```diff
@@ -0,0 +1,88 @@
+import { Doc, Path, WorkspaceAddress } from "../util/doc-types";
+import { StorageDriverAsyncMemory } from "./storage-driver-async-memory";
+
+type SerializedDriverDocs = {
+  byPathAndAuthor: Map<string, Doc>;
+  byPathNewestFirst: Map<Path, Doc[]>;
+};
+
+function isSerializedDriverDocs(value: any): value is SerializedDriverDocs {
+  if (typeof value !== "object") {
+    return false;
+  }
+
+  if (
+    "byPathAndAuthor" in value &&
+    "byPathNewestFirst" in value &&
+    value.byPathAndAuthor instanceof Map &&
+    value.byPathNewestFirst instanceof Map
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+export class StorageDriverLocalStorage extends StorageDriverAsyncMemory {
+  _localStorageKeyConfig: string;
+  _localStorageKeyDocs: string;
+
+  constructor(workspace: WorkspaceAddress) {
+    super(workspace);
+
+    this._localStorageKeyConfig = `earthstar:config:${workspace}`;
+    this._localStorageKeyDocs = `earthstar:documents:pathandauthor:${workspace}`;
+
+    let existingData = localStorage.getItem(this._localStorageKeyDocs);
+
+    if (existingData !== null) {
+      let parsed = JSON.parse(existingData);
+
+      if (!isSerializedDriverDocs(parsed)) {
+        return;
+      }
+
+      this.docByPathAndAuthor = parsed.byPathAndAuthor;
+      this.docsByPathNewestFirst = parsed.byPathNewestFirst;
+    }
+  }
+
+  async getConfig(key: string): Promise<string | undefined> {
+    key = `${this._localStorageKeyConfig}:${key}`;
+    let result = localStorage.getItem(key);
+    return result === null ? undefined : result;
+  }
+  
+  async setConfig(key: string, value: string): Promise<void> {
+    super.setConfig(key, value);
```

Comment:
Any reason to use the superclass here when we could just replace it with a localStorage implementation?

Maybe so we can use the `listConfigKeys` [from the superclass](https://github.com/earthstar-project/stone-soup/blob/main/src/storage/storage-driver-async-memory.ts#L107-L109)?

Reply:
Yes, that was the idea! I'm not sure if this is the best arrangement of these methods.